### PR TITLE
HPCC-21397 Docs: Improve IFF() description

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-IFF.xml [hunk 1]
```diff
@@ -50,11 +50,26 @@
...
-   <para>The <emphasis role="bold">IFF</emphasis> function performs the same
+  <para>The <emphasis role="bold">IFF</emphasis> function performs the same
   functionality as IF, but ensures that an <emphasis>expression</emphasis>
   containing complex boolean logic is evaluated exactly as it appears.</para>
 
-  <para>See Also: <link linkend="IF">IF</link>, <link linkend="MAP">MAP</link>,
-  <link linkend="EVALUATE">EVALUATE</link>, <link linkend="CASE">CASE</link>,
-  <link linkend="CHOOSE">CHOOSE</link>, <link linkend="SET">SET</link></para>
+  <para>The main difference between IF and IFF is with conditional scalar
+  expressions. Generally the arguments to dataset IF() actions are always
```
Comment (ghalliday) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-IFF.xml:58]: minor: change actions->operators

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-IFF.xml [hunk 2]
```diff
@@ -50,11 +50,26 @@
...
+  evaluated lazily, but there are situations for rows and scalars where they
```
Comment (ghalliday) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-IFF.xml:59]: possibly clarify
`there are situations for rows and scalars`
to
`there are situations, if the expression type is a row or a scalar,`

File: docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-IFF.xml [hunk 3]
```diff
@@ -50,11 +50,26 @@
...
+  might not be.</para>
+
+  <para>For example:</para>
+
+  <para><programlisting>foo := IF (age &lt; 18, '', expensiveFunction());
+</programlisting></para>
+
+  <para>Then there is no guarantee that expensiveFunction will not be
+  executed. Using IFF() converts it to a dataset expression, ensuring it is
```
Comment (ghalliday) [inline docs/EN_US/ECLLanguageReference/ECLR_mods/BltInFunc-IFF.xml:68]: Possibly
`Using IFF() converts`
to
`If IFF() is used the code generator internally converts`

**Review by ghalliday — COMMENTED** [review]
@JamesDeFabia a few suggested tweaks.

**Review by ghalliday — APPROVED** [review]
@JamesDeFabia please squash and I'll merge

**Comment by JamesDeFabia** [comment]
@ghalliday Squashed

**Pull request merged**